modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenOperation.java

```diff
@@ -28,6 +28,7 @@
     public List<CodegenParameter> queryParams = new ArrayList<CodegenParameter>();
     public List<CodegenParameter> headerParams = new ArrayList<CodegenParameter>();
     public List<CodegenParameter> formParams = new ArrayList<CodegenParameter>();
+    public List<CodegenParameter> requiredParams = new ArrayList<>();
```

Comment:
I think you need to put `CodegenParameter` in the ArrayList to be able to compile swagger-codegen in Java < 1.7

Reply:
Will do right now! Yes, makes sense to support 1.6.